### PR TITLE
New profile pictures

### LINKED ISSUE #170: Redesign sidebar
Highlight which page we're on. 
Change color scheme.

IMG_2099

### LINKED ISSUE #171: Different profile picture
Instead of Henry, we can have generic user icon with the first letter of displayName shown. 
Or just generic user icon, we'll see.

### PR DESCRIPTION
**Pull request by shchau**
resolves #171 
resolves #170 

Profile picture changed to generic user icon with the first letter of displayName in it. 

Recreated sidebar to use semantic ui react, highlights current tab now. Adjusted margins for friends cards.
Also added "Failed to get posts" message for posts. 

author
friends
public
stream
failed

### CODE REVIEW

File: frontend/src/components/CommentsOnPost.jsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import HTTPFetchUtil from "../util/HTTPFetchUtil";
+import PropTypes from 'prop-types';
+import './styles/CommentsOnPost.css';
+
+class CommentsOnPost extends Component {	
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			comments: [],
```
Comment (derrick-w) [inline frontend/src/components/CommentsOnPost.jsx]: this file is part of your old comments PR?

File: frontend/src/components/CommentsOnPost.jsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		        console.error(error);
+			});		
+	}
+	
+	createCommentList(comment) {
+		const commentID = comment.id;
+		const author = comment.author.displayName;
+		const authorID = comment.author.id;
+		const date = comment.published;
+		const commentText = comment.comment;
```
Comment (derrick-w) [inline frontend/src/components/CommentsOnPost.jsx]: probably do this in 1 const?

File: frontend/src/components/CommentsOnPost.jsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	}
+
+	handleChange(event) {
+		this.setState({[event.target.name]: event.target.value});
+	}
+	
+	
+	submitComment() {
+		const urlPath = "/api/posts/" + this.props.postID + "/comments";
+		const requireAuth = true;
```
Comment (derrick-w) [inline frontend/src/components/CommentsOnPost.jsx]: could do this in 1 const

**Comment by derrick-w** [comment]
is it possible for profile picture bubbles to have capital letters

File: frontend/src/components/CommentsOnPost.jsx (hunk 1 above)
Comment (shchau) [reply frontend/src/components/CommentsOnPost.jsx]: o oops, I'll take this file and the css file out.

**Comment by shchau** [comment]
> is it possible for profile picture bubbles to have capital letters

Yes but it's ugly.

**Comment by hiufungk** [comment]
change the friend request, current friends button to have consistent colour scheme
change the follow, unfollow button button to be consistent with the author page
also not a fan of that nav bar colour when it is selected 
image

File: frontend/src/components/StreamFeed.jsx [hunk 4]
```diff
@@ -140,7 +146,11 @@ class StreamFeed extends Component {
 		return(	
 		<div>
 			<Feed>
-				<Loader active={this.state.isFetching}/>
+				{this.state.isFetching && <Loader/>}
+				{this.state.error && <Message className="failedGetPostsMessage" negative>
```
Comment (hiufungk) [inline frontend/src/components/StreamFeed.jsx:273]: use the disabled prop for Loader
use the hidden prop for Message

**Review by forgeno — APPROVED** [review]

File: frontend/src/pages/Stream.jsx [hunk 5]
```diff
@@ -9,7 +9,11 @@ class Stream extends Component {
 		const storeItems = store.getState().loginReducers;
 		return(	
 			<div className="pusher">
+<<<<<<< HEAD
```
Comment (forgeno) [inline frontend/src/pages/Stream.jsx]: can you fix these merge conflicts

**Review by forgeno — CHANGES_REQUESTED** [review]

**Review by forgeno — APPROVED** [review]

**Review by derrick-w — APPROVED** [review]
image

**Pull request merged**